### PR TITLE
Add spring watch

### LINKED ISSUE #81: Trouble with spring after moving / renaming packages?
We're running into situations where `packwerk check` or `packs check` returns incorrect results in some situations, like after changing branches. I believe that this is due to an interaction (or lack thereof) between `packs-rails` and `spring`.

Is anyone else seeing this problem?

I just (seemingly) fixed this on a project I am working on.

It looks like the explanation is the following:

`packs-rails` sets up autoload paths by detecting `package.yml`s. So moving packs around, renaming them, deleting packs or creating new ones leads to the autoload paths being changed. The application has to be restarted for that. Which means `spring` needs to watch all the `package.yml`s. I just did a `Dir["packs/**/package.yml"],each { |f| Spring.watch(f) }` to fix it for us.

But if this is the actual explanation, everybody setting up `packs-rails` naively should have this problem, and I haven't been hearing much about it. Does this ring a bell for anyone?

If this is actually a common problem, maybe we can have `packs-rails` set up the spring watches implicitly?

### PR DESCRIPTION
**Pull request by professor**
When a pack moves, we want spring to update the loaded classes so that packwerk prints correct results

Screenshot 2024-02-13 at 13 40 43

Fixes #81

### CODE REVIEW

File: lib/packs/rails/railtie.rb [hunk 1]
```diff
@@ -11,7 +11,19 @@ class Railtie < ::Rails::Railtie
...
         ActiveSupport.run_load_hooks(:packs_rails, Packs)
       end
-      
+
+      config.after_initialize do |app|
+        if defined?(Spring)
+          puts "Hello Db Blasters!"
+          Packs::Specification::Configuration.fetch.pack_paths.each do |dir|
+            puts "#{dir}"
+            Dir["#{dir}/package.yml"].each { |f|
```
Comment (exterm) [inline lib/packs/rails/railtie.rb:20]: why would you do a glob here (`Dir[]`)? Isn't `dir` the exact path? In which case you'll want `File.join(dir, "package.yml")` or something like that

But I'm probably misunderstanding `pack_paths`...

File: lib/packs/rails/railtie.rb [hunk 2]
```diff
@@ -11,7 +11,15 @@ class Railtie < ::Rails::Railtie
         # hook into packs-rails via ActiveSupport hooks.
         ActiveSupport.run_load_hooks(:packs_rails, Packs)
       end
-      
+
+      config.after_initialize do |app|
```
Comment (ngan) [inline lib/packs/rails/railtie.rb:15]: Any reason why this has to be in an after initialize a not a normal initializer?

Comment (perryqh) [reply lib/packs/rails/railtie.rb:15]: Great question!

Are you suggesting that apps including this gem should set up `Spring.watch` for packwerk within their own config/initializers file?

Considering the [available options](https://api.rubyonrails.org/classes/Rails/Railtie/Configuration.html), using after_initialize seems logical as it ensures Spring is loaded before this block is ran, provided it's defined.

File: lib/packs/rails/railtie.rb (hunk 1 above)
Comment (perryqh) [reply lib/packs/rails/railtie.rb]: Great question!

Example `pack_paths`:
- packs/*
- packs/*/*

Comment (exterm) [reply lib/packs/rails/railtie.rb]: ah, makes sense!

Comment (exterm) [reply lib/packs/rails/railtie.rb]: I think `pack_paths` is easy to misunderstand; it's more like `pack_path_specs` or `pack_path_globs`, but that's out of scope here

**Comment by exterm** [comment]
Very excited for this to land

File: lib/packs/rails/railtie.rb (hunk 2 above)
Comment (ngan) [reply lib/packs/rails/railtie.rb:15]: I think this should be either before_configuration or just a regular initializer

**Comment by exterm** [comment]
One note: this doesn't seem to cover the case where an autoload path is added and none of the previously registered `package.yml` files is changed.

I just added an initializer to a pack and had to stop spring manually.

**Comment by perryqh** [comment]
> One note: this doesn't seem to cover the case where an autoload path is added and none of the previously registered `package.yml` files is changed.
> 
> I just added an initializer to a pack and had to stop spring manually.

Thanks for the comment!

We want to clarify the expectation. Did you add a new autoload path or a new file _in_ the autoload path?

We noticed that Spring has a limitation on only watching files it knows about when initialized. New files, including ruby files, will not cause Spring to reload.

**Comment by professor** [comment]
Replaced by https://github.com/rubyatscale/packs-rails/pull/86

**Pull request closed without merging**